Extensions/TextInput/textinputruntimeobject-pixi-renderer.ts

```diff
@@ -0,0 +1,201 @@
+    private _object: gdjs.TextInputRuntimeObject;
+    private _input: HTMLInputElement | HTMLTextAreaElement;
+    private _runtimeScene: gdjs.RuntimeScene;
+    private _runtimeGame: gdjs.RuntimeGame;
+
+    constructor(runtimeObject: gdjs.TextInputRuntimeObject) {
+      this._object = runtimeObject;
+      this._runtimeScene = runtimeObject.getRuntimeScene();
+      this._runtimeGame = this._runtimeScene.getGame();
+
+      this._input = this._createElement();
+      this._runtimeGame
+        .getRenderer()
+        .getDomElementContainer()!
+        .appendChild(this._input);
+    }
+
+    _createElement() {
+      const isTextArea = this._object.getInputType() === 'text area';
+      this._input = document.createElement(isTextArea ? 'textarea' : 'input');
+      this._input.style.border = '1px solid black';
+      this._input.style.borderRadius = '0px';
+      this._input.style.backgroundColor = 'white';
+      this._input.style.position = 'absolute';
+      this._input.style.resize = 'none';
+      this._input.style.pointerEvents = 'auto'; // Element can be clicked/touched.
+
+      this._input.addEventListener('input', () => {
+        this._object.onRendererInputValueChanged(this._input.value);
+      });
+      this._input.addEventListener('touchstart', () => {
+        // Focus directly when touching the input on touchscreens.
+        if (document.activeElement !== this._input) this._input.focus();
+      });
+
+      this.updateValue();
+      this.updateFont();
+      this.updatePlaceholder();
+      this.updateOpacity();
+      this.updateInputType();
+      this.updateTextColor();
+      this.updateFillColorAndOpacity();
+      this.updateBorderColorAndOpacity();
+
+      return this._input;
+    }
+
+    _destroyElement() {
+      if (!this._input) return;
+
+      const parentElement = this._input.parentElement;
+      if (parentElement) parentElement.removeChild(this._input);
+    }
+
+    onDestroy() {
+      this._destroyElement();
+    }
+
+    updatePreRender() {
+      const layer = this._runtimeScene.getLayer(this._object.getLayer());
+      const runtimeGame = this._runtimeScene.getGame();
+      const runtimeGameRenderer = runtimeGame.getRenderer();
+      const topLeftCanvasCoordinates = layer.convertInverseCoords(
+        this._object.x,
+        this._object.y,
+        0
+      );
+      const bottomRightCanvasCoordinates = layer.convertInverseCoords(
+        this._object.x + this._object.getWidth(),
+        this._object.y + this._object.getHeight(),
+        0
+      );
+
+      // Hide the input entirely if not visible at all
+      const isOutsideCanvas =
+        bottomRightCanvasCoordinates[0] < 0 ||
+        bottomRightCanvasCoordinates[1] < 0 ||
+        topLeftCanvasCoordinates[0] > runtimeGame.getGameResolutionWidth() ||
+        topLeftCanvasCoordinates[1] > runtimeGame.getGameResolutionHeight();
+      if (isOutsideCanvas) {
+        this._input.style.display = 'none';
```

Comment:
Not tested, this was mostly an hypothesis. I'll take a look at if we can entirely remove these from the DOM, with a safety margin to avoid perpetually add/remove stuff from the DOM if you are at the limit where the object could be created/destroyed.